legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/UnitemporalSnapshotPlanner.java

```diff
@@ -53,6 +53,12 @@ class UnitemporalSnapshotPlanner extends UnitemporalPlanner
         if (ingestMode.partitioned())
         {
             List<String> fieldNames = stagingDataset().schema().fields().stream().map(Field::name).collect(Collectors.toList());
+            // All partitionFields must be present in staging dataset
+            ingestMode.partitionFields().forEach(field -> validateExistence(
+                    fieldNames,
+                    field,
+                    "Field [" + field + "] from partitionFields not present in incoming dataset"));
+            // All partitionValuesByField must be present in staging dataset
             ingestMode.partitionValuesByField().keySet().forEach(field -> validateExistence(
                 fieldNames,
                 field,
```

Comment:
Actually, after adding line 56 to line 60, the next few lines (62 - 65) become unnecessary as we have previously validated that the fields in partitionFields and the fields in the keyset of partitionValuesByField are the same

Reply:
That's right, will remove it